qiime2/core/cache.py

```diff
@@ -761,7 +762,19 @@ def save(self, ref, key):
 
         return self.load(key)
 
-    def _register_key(self, key, value, pool=False):
+    def save_collection(self, ref_collection, key):
+        """Saves a Collection to a pool in the cache with the given key. This
+        pool's key file will keep track of the order of the Collection.
+        """
+        with self.lock:
+            pool = self._create_collection_pool(ref_collection, key)
+
+            for ref in list(ref_collection.values()):
+                pool.save(ref)
```

Comment:
This could take a while, do we need to do something similar to `_rename_to_data` so that we aren't locking a cache for the duration it takes a collection to be (re-)written to a cache?

Reply:
pool.save is already pretty efficient, it creates a symlink in the pool then calls Cache._rename_to_data